backend/src/main/java/ca/gov/dtsstn/cdcp/api/service/UserService.java

```diff
@@ -134,4 +134,25 @@ private Predicate<ConfirmationCodeEntity> byCode(String code) {
 		return confirmationCode -> confirmationCode.getCode().equals(code);
 	}
 
+	private Predicate<ConfirmationCodeEntity> notExpired() {
+		return confirmationCode -> confirmationCode.getExpiryDate().isAfter(Instant.now());
+	}
+
+    public Boolean verifyConfirmationCode(String code, User user){
+		Assert.hasText(code, "code is required; it must not be null or blank");
+		Assert.notNull(user, "user is required; it must not be null");
+		if (user.getConfirmationCodes().stream().map(confirmationCodeMapper::toEntity).filter(notExpired()).noneMatch(byCode(code))) {
+    		return false;
+    	}
+        return true;
```

Comment:
When a confirmation code is validated, all confirmation codes for the user must be deleted from the database and the user's `emailValidated` flag must be set to `true`.